nestkernel/source_table.cpp

```diff
@@ -320,67 +364,59 @@ nest::SourceTable::get_next_target_data( const thread tid,
     current_source.set_processed( true );
 
     // we need to set a marker stating whether the entry following this
-    // entry, if existent, has the same source; start by assuming it
-    // has a different source, only change if necessary
-    kernel().connection_manager.set_has_source_subsequent_targets(
-      current_position.tid, current_position.syn_id, current_position.lcid, false );
-    if ( ( current_position.lcid + 1
-             < static_cast< long >( sources_[ current_position.tid ][ current_position.syn_id ].size() )
-           and sources_[ current_position.tid ][ current_position.syn_id ][ current_position.lcid + 1 ].get_node_id()
-             == current_source.get_node_id() ) )
+    // entry, if existent, has the same source
+    kernel().connection_manager.set_has_source_subsequent_targets( current_position.tid,
+      current_position.syn_id,
+      current_position.lcid,
+      next_entry_has_same_source( current_position, current_source ) );
+
+    // no need to communicate this entry if the previous entry has the
+    // same source
```

Comment:
Since we allow long lines now, I'd put this comment on one line.

Reply:
ok